src/tests/ftest/recovery/pool_membership.py

```diff
@@ -0,0 +1,205 @@
+        :avocado: tags=PoolMembershipTest,test_orphan_pool_shard
+        """
+        # 1. Create a pool on rank 0.
+        pool = self.get_pool(connect=False, target_list="0")
+
+        # 2. Call dmg storage query usage to store the default space utilization.
+        rank_to_free_orig = self.get_rank_to_free()
+        self.log.info("rank_to_free_orig = %s", rank_to_free_orig)
+
+        # 3. Prepare to copy the pool path before stopping servers.
+
+        # In order to copy the pool directory without password, there are several things
+        # to determine and set up.
+
+        # 3-1. Determine source host and destination host. Source host is where rank 0 is.
+        # Destination host is the other host.
+        src_host = dst_host = NodeSet(self.server_managers[0].get_host(0))
+        rank = 1
+        while rank < self.server_managers[0].engines and dst_host == src_host:
+            dst_host = NodeSet(self.server_managers[0].get_host(rank))
+            rank += 1
+        self.log.info("src_host = %s; dst_host = %s", src_host, dst_host)
+
+        # 3-2. Determine source and destination mount point. First, find the source mount
+        # point that maps to rank 0. Then use the same mount point for the destination.
+        # This way, we can handle the case where the mount point name is changed in the
+        # future. At the same time, determine the destination rank, which is where
+        # dst_mount is mapped.
+        src_mount = None
+        dst_mount = None
+        dst_rank = None
+        dmg_command = self.get_dmg_command()
+        storage_query_out = dmg_command.storage_query_usage()
+        hash_dict = storage_query_out["response"]["HostStorage"]
+        for storage_dict in hash_dict.values():
+            if str(src_host) in storage_dict["hosts"]:
+                # Determine source mount point that maps to rank 0.
+                for scm_namespace in storage_dict["storage"]["scm_namespaces"]:
+                    if scm_namespace["mount"]["rank"] == 0:
+                        # For dst_mount, use the same mount point as source.
+                        dst_mount = src_mount = scm_namespace["mount"]["path"]
+        for storage_dict in hash_dict.values():
+            if str(dst_host) in storage_dict["hosts"]:
+                # Determine destination rank that maps to dst_mount.
+                for scm_namespace in storage_dict["storage"]["scm_namespaces"]:
+                    if scm_namespace["mount"]["path"] == dst_mount:
+                        dst_rank = scm_namespace["mount"]["rank"]
+
+        # 4. Stop servers.
+        self.server_managers[0].system_stop()
+
+        # 5. Copy /mnt/daos?/<pool_path> from the engine where we created the pool to
+        # another engine where we didn’t create.
+
+        # 5-1. Since we're running rsync as user, update the mode of the source pool
+        # directory to 777.
+        self.log.info("Update mode of the source pool directory.")
```

Comment:
These probably should be `self.log_step('test doing something')` calls, but then you'd want to add additional ones for earlier steps, like `self.log_step('Creating a pool (dmg pool create)')`, `self.log_step('Collecting free space for each rank (dmg storage query usage)')`, and `self.log_step('Determine source host and destination host')`

Reply:
I agree we should start using `self.log_step` for new tests, but I think we also need to mention this in the Test WG so everyone is aware

Reply:
Done